server/core/client-manager.js

```diff
@@ -0,0 +1,32 @@
+const EventEmitter = require('events');
+const Client = require('./client');
+
+module.exports = class ClientManager extends EventEmitter {
```

Comment:
Yes, this one is kind of tricky and had a few ways to be implemented.

1) In the current implementation the `ClientManager` class is an emitter (extends `EventEmitter`), because it has to emit the `new-client` event to the Radio. It currently does not use any other events from the emitter. In addition to this it includes a `socketIo` instance, which is an emitter (what you described above) and it redirects events to it with the `emitToAll` method, it also listens on that emitter for connecting/disconnecting users. I'll make sure to describe this in the future documentation. :)

2) The other possible approach here was to use only `socketIo`'s emitter for all events and not make our class an `EventEmitter`. I could've implemented `emit` and `on` methods that would only redirect to socketIo's emitter. However I wanted to:
    * Capture `emit` calls and redirect them to `socketIo`
    * Emit a `new-client` event, which the Radio can catch and use.
In this case if I emit a `new-client` and use `socketIo`'s emitter I would also catch it and have to redirect it to `socketIo`, which wouldn't make any sense. 

Instead of making exceptional cases for events which won't be redirected to `socketIo` in `2)` I separated it into two emitters with approach `1)`. Downside is that I can't use an `emit` method to redirect to socket.io (as it's already declared/used by extending `EventEmitter`) so it was called `emitToAll` instead.